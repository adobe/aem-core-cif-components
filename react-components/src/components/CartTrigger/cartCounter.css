.root {
    font-weight: 600;
    margin-left: 0.3rem;
}
