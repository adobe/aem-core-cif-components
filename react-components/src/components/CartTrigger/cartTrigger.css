.root {
    composes: root from '../clickable.css';
    height: 3rem;
    min-width: 3rem;
}
