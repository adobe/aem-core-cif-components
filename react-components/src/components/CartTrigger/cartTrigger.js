/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import PropTypes from 'prop-types';
import ReactDOM from 'react-dom';

import CartCounter from './cartCounter';

import Icon from '../Icon';
import { ShoppingCart as ShoppingCartIcon } from 'react-feather';
import classes from './cartTrigger.css';
import { useCartState } from '../Minicart/cartContext';

const parentEl = document.querySelector('.header__cartTrigger');

const Trigger = () => {
    const [{ cart }, dispatch] = useCartState();
    let cartQuantity = cart && Object.entries(cart).length > 0 ? cart.items.length : 0;

    const iconColor = 'rgb(var(--venia-text))';
    const svgAttributes = {
        stroke: iconColor
    };

    if (cartQuantity > 0) {
        svgAttributes.fill = iconColor;
    }

    const button = (
        <button className={classes.root} aria-label="Toggle mini cart" onClick={() => dispatch({ type: 'open' })}>
            <Icon src={ShoppingCartIcon} attrs={svgAttributes} />
            <CartCounter counter={cartQuantity} />
        </button>
    );

    return ReactDOM.createPortal(button, parentEl);
};

Trigger.propTypes = {
    children: PropTypes.node,
    classes: PropTypes.shape({
        root: PropTypes.string
    })
};

export default Trigger;
