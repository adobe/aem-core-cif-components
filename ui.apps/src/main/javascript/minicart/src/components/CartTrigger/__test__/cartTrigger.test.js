/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import ReactDOM from 'react-dom';
import { render, fireEvent } from '@testing-library/react';
import CartTrigger from '../cartTrigger';

describe('<CartTrigger>', () => {
    beforeAll(() => {
        ReactDOM.createPortal = jest.fn((element, node) => {
            return element;
        });
    });

    afterEach(() => {
        ReactDOM.createPortal.mockClear();
    });

    it('renders the icon', () => {
        const { asFragment } = render(<CartTrigger cartQuantity={2} handler={jest.fn()} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the quantity', () => {
        const expectedQuantity = '2';
        const { getByTestId } = render(<CartTrigger cartQuantity={parseInt(expectedQuantity)} handler={jest.fn()} />);

        expect(getByTestId('cart-counter').textContent).toEqual(expectedQuantity);
    });

    it('calls the handler function when clicked', () => {
        const handler = jest.fn();

        const { getByRole } = render(<CartTrigger cartQuantity={2} handler={handler} />);
        fireEvent.click(getByRole('button'));

        expect(handler.mock.calls.length).toEqual(1);
    });
});
