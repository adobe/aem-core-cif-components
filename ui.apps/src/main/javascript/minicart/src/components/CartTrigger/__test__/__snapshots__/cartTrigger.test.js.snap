// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartTrigger> renders the icon 1`] = `
<DocumentFragment>
  <button
    aria-label="Toggle mini cart"
    class="root"
  >
    <span
      class="root"
    >
      <svg
        fill="rgb(var(--venia-text))"
        height="24"
        stroke="rgb(var(--venia-text))"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="9"
          cy="21"
          r="1"
        />
        <circle
          cx="20"
          cy="21"
          r="1"
        />
        <path
          d="M1 1h4l2.68 13.39a2 2 0 0 0 2 1.61h9.72a2 2 0 0 0 2-1.61L23 6H6"
        />
      </svg>
    </span>
    <span
      class="root"
      data-testid="cart-counter"
    >
      2
    </span>
  </button>
</DocumentFragment>
`;
